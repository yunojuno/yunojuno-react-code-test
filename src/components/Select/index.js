import React from 'react';

export default function Select ({ value, callback, options }) {
  return (
    <select value={value} onChange={callback} >
      {options.map(el => (
        <option key={el} value={el}>{el.charAt(0).toUpperCase() + el.slice(1)}</option>
      ))}
    </select>
  );
}
