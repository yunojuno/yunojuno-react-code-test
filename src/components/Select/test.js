import React from "react";
import { shallow } from 'enzyme';

import Select from './index';

const TEST_ARRAY = ["1", "2", "3"];

const select = shallow(<Select options={TEST_ARRAY} value="1" />);

it("renders correctly", () => {
  expect(select).toMatchSnapshot();
});

it('displays a reasonable value', () => {
  expect(TEST_ARRAY).toContain(select.prop('value'));
});
