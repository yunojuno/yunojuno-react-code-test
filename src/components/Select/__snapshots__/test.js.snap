// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Select
    options={
        Array [
            "1",
            "2",
            "3",
          ]
    }
    value="1"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <option
          value="1"
>
          1
</option>,
        <option
          value="2"
>
          2
</option>,
        <option
          value="3"
>
          3
</option>,
      ],
      "onChange": undefined,
      "value": "1",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "1",
          "value": "1",
        },
        "ref": null,
        "rendered": "1",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": "2",
          "value": "2",
        },
        "ref": null,
        "rendered": "2",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": "3",
          "value": "3",
        },
        "ref": null,
        "rendered": "3",
        "type": "option",
      },
    ],
    "type": "select",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            value="1"
>
            1
</option>,
          <option
            value="2"
>
            2
</option>,
          <option
            value="3"
>
            3
</option>,
        ],
        "onChange": undefined,
        "value": "1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "1",
            "value": "1",
          },
          "ref": null,
          "rendered": "1",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "2",
            "value": "2",
          },
          "ref": null,
          "rendered": "2",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": "3",
            "value": "3",
          },
          "ref": null,
          "rendered": "3",
          "type": "option",
        },
      ],
      "type": "select",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
